lib/Coverage.js

```diff
@@ -127,6 +139,104 @@ class JSCoverage {
   }
 }
 
+class CSSCoverage {
+  /**
+   * @param {!Puppeteer.Session} client
+   */
+  constructor(client) {
+    this._client = client;
+    this._enabled = false;
+    this._stylesheetURLs = new Map();
+    this._stylesheetSources = new Map();
+    this._eventListeners = [];
+    this._resetOnNavigation = false;
+  }
+
+  /**
+   * @param {!Object} options
+   */
+  async start(options = {}) {
+    console.assert(!this._enabled, 'CSSCoverage is already enabled');
+    this._resetOnNavigation = options.resetOnNavigation === undefined ? true : !!options.resetOnNavigation;
+    this._enabled = true;
+    this._stylesheetURLs.clear();
+    this._stylesheetSources.clear();
+    this._eventListeners = [
+      helper.addEventListener(this._client, 'CSS.styleSheetAdded', this._onStyleSheet.bind(this)),
+      helper.addEventListener(this._client, 'Runtime.executionContextsCleared', this._onExecutionContextsCleared.bind(this)),
+    ];
+    await Promise.all([
+      this._client.send('DOM.enable'),
+      this._client.send('CSS.enable'),
+      this._client.send('CSS.startRuleUsageTracking'),
+    ]);
+  }
+
+  _onExecutionContextsCleared() {
+    if (!this._resetOnNavigation)
+      return;
+    this._stylesheetURLs.clear();
+    this._stylesheetSources.clear();
+  }
+
+  /**
+   * @param {!Object} event
+   */
+  async _onStyleSheet(event) {
+    const header = event.header;
+    // Ignore anonymous scripts
+    if (!header.sourceURL)
+      return;
+    try {
+      const response = await this._client.send('CSS.getStyleSheetText', {styleSheetId: header.styleSheetId});
+      this._stylesheetURLs.set(header.styleSheetId, header.sourceURL);
+      this._stylesheetSources.set(header.styleSheetId, response.text);
+    } catch (e) {
+      // This might happen if the page has already navigated away.
+      debugError(e);
+    }
+  }
+
+  /**
+   * @return {!Promise<!Array<!{url:string, text:string, ranges:!Array<!{start:number, end:number}>}>>}
```

Comment:
not: can we have this typedef as CoverageData

Reply:
nit* To force the same format

Reply:
Done